docs/rfcs/1-command-line-design.md

```diff
@@ -0,0 +1,63 @@
+- Author: xiongjiwei <xiongjiwei1996@outlook.com>
+- Start Date: 2021-08-29
+- RFC PR: N/A
+- Tracking Issue: [beyondstorage/beyond-ftp#12](https://github.com/beyondstorage/beyond-ftp/issues/12)
+
+# RFC-1: Command Line Design
+
+## Background
+
+Command line is a interface to use the software, user can use it to control the software's behavior.
+
+## Proposal
+
+A command line program include the `commands`, `flags` and `args`. For Beyond-FTP, they are:
+
+- subcommands
+
+| name    | usage                      |
+|---------|----------------------------|
+| version | show beyond-ftp version    |
+| help    | show usage of beyond-ftp   |
+
+These two subcommands does not have any `flags` or `args`.
+
+- flags
+
+| name   | shorthand | default   | type   | require | usage                        |
+|--------|-----------|-----------|--------|---------|------------------------------|
+| config | c         |           | string | N       | config file path             |
+| host   |           | 127.0.0.1 | string | N       | server listen host           |
+| port   | p         | 21        | number | N       | server listen port           |
+| debug  | d         | false     | bool   | N       | start server with debug mode |
+
+`config` provide a config file to start FTP server, if no file is specified, the configures will use its default value.
+
+`host` and `port` provide the host and port that the server should listend. The default value is `127.0.0.1:21`.
+
+`debug` indicate the server start with debug mode. Debug mode will print log with debug level and provide profiles, and use memory as under storage.
+
+- args
+
+N/A
+
+- config format
+
+use `toml` as the config format, it is easy for human read.
+
+- apply order

```

Comment:
Maybe we can support env values?

So that we can use `BEYOND_FTP_ADDR=127.0.0.1:21 beyondftp` without config. It's useful for running in the container.

Reply:
I think `flags` is enough? I am not familiar with the container environment, could you please give some advice?

Reply:
Usually, we can't have a config file to read if we start from a container, users have to `mount` a config file into it. And it's more complex if running in K8s. So we usually provide env support so that users can config all things via env instead of config file or flags.

For example: minio: https://docs.min.io/minio/baremetal/reference/minio-server/minio-server.html#id6

Maybe we should discuss it along with the config format spec..